Cython/Compiler/ExprNodes.py

```diff
@@ -655,6 +655,19 @@ def analyse_as_type(self, env):
         # type, return that type, else None.
         return None
 
+    def analyse_as_specialized_type(self, env):
+        type = self.analyse_as_type(env)
+        if type.is_fused and env.fused_to_specific:
+            # while it would be nice to test "if entry.type in env.fused_to_specific"
+            # rather than try/catch this doesn't work reliably (mainly for nested fused types)
+            try:
```

Comment:
I know this `try:` was pointed out before in a previous iteration of this PR. It does seem necessary, unfortunately, and it isn't the only place where this pattern's used (suggesting people have found it necessary before) https://github.com/cython/cython/blob/04d918f684404375390657d12ecb53346d66c93d/Cython/Compiler/ParseTreeTransforms.py#L3344-L3349